src/test/java/com/softserve/academy/spaced/repetition/service/AccessToUrlServiceTest.java

```diff
@@ -0,0 +1,283 @@
+package com.softserve.academy.spaced.repetition.service;
+
+import com.softserve.academy.spaced.repetition.domain.*;
+import com.softserve.academy.spaced.repetition.domain.enums.AuthorityName;
+import com.softserve.academy.spaced.repetition.repository.*;
+import com.softserve.academy.spaced.repetition.service.impl.AccessToUrlServiceImpl;
+import com.softserve.academy.spaced.repetition.util.DomainFactory;
+import com.softserve.academy.spaced.repetition.utils.exceptions.NotAuthorisedUserException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+@Transactional
+public class AccessToUrlServiceTest {
+
+    private final Long CATEGORY_ID = 1L;
+    private final Long COURSE_ID = 1L;
+    private final Long CARD_ID = 1L;
+    private final Long FOLDER_ID = 1L;
+    private final Long DECK_ID = 1L;
+    private final Long COMMENT_ID = 1L;
+    private Category category;
+    private List<Course> courseList;
+    private List<Deck> deckList;
+
+    @Mock
+    private CategoryRepository categoryRepository;
+
+    @Mock
+    private UserService userService;
+
+    @Mock
+    private CourseRepository courseRepository;
+
+    @Mock
+    private FolderRepository folderRepository;
+
+    @Mock
+    private DeckRepository deckRepository;
+
+    @Mock
+    private CardRepository cardRepository;
+
+    @Mock
+    private DeckCommentRepository deckCommentRepository;
+
+    @Mock
+    private CourseCommentRepository courseCommentRepository;
+
+    @InjectMocks
+    private AccessToUrlServiceImpl accessToUrlService;
+
+    @Before
+    public void setUp() throws Exception {
+        courseList = new ArrayList<>();
+        deckList = new ArrayList<>();
+        List<Card> cardList = new ArrayList<>();
+        User user = DomainFactory.createUser(1L, new Account(), new Person(), new Folder(), null);
```

Comment:
Maybe will be better to create `final` vairables instead of 'magic strings' and 'magic numbers'

Reply:
I`m not sure that this additional code lines will be usefull.

Reply:
Maybe, in future, you will need to change some values and you will not need to scroll all of this code.